libs/web/shell/ui/album-art-overlay/src/lib/album-art-overlay.component.ts

```diff
@@ -1,23 +1,55 @@
 import { PlaybackStore } from '@angular-spotify/web/shared/data-access/store';
-import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { map } from 'rxjs/operators';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  NgZone,
+  ViewChild
+} from '@angular/core';
+import { filter, map } from 'rxjs/operators';
 
 @Component({
   selector: 'as-album-art-overlay',
   templateUrl: './album-art-overlay.component.html',
   styleUrls: ['./album-art-overlay.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class AlbumArtOverlayComponent {
-  albumBackgroundImage$ = this.playbackStore.currentTrack$.pipe(
-    map((track) => {
-      if (!track?.album?.images) {
-        return null;
+export class AlbumArtOverlayComponent implements AfterViewInit {
+  @Input() set imageUrl(url: string) {
+    this.zone.runOutsideAngular(() => {
+      this.drawImageToCanvas(url);
+    });
+  }
+
+  @ViewChild('canvas') canvas!: ElementRef<HTMLCanvasElement>;
+  context!: CanvasRenderingContext2D | null;
+
+  constructor(private zone: NgZone) {}
+
+  ngAfterViewInit() {
+    this.zone.runOutsideAngular(() => {
+      this.initCanvas();
+    });
+  }
+
+  drawImageToCanvas(imageUrl: string) {
+    const imageObj = new Image();
+    imageObj.src = imageUrl;
+    imageObj.onload = () => {
+      if (this.context) {
+        this.context.clearRect(0, 0, this.context.canvas.width, this.context.canvas.height);
+        this.context.drawImage(imageObj, 0, 0);
       }
-      const imageUrl = track.album.images[0]?.url;
-      return imageUrl ? `url(${imageUrl})` : null;
-    })
-  );
+    };
+  }
 
-  constructor(private playbackStore: PlaybackStore) {}
+  initCanvas() {
+    this.context = this.canvas.nativeElement.getContext('2d');
+    if (this.context) {
```

Comment:
nitpick: this if check looks like it's redundant.

Reply:
Noted, so we can just put `!` in `this.context!.`, right?

Reply:
`context!: CanvasRenderingContext2D | null;` I didn't see this. this is redundant as well. if you plan to use `!` non-null assertion then might as well removing the `null` type from this.